src/models/NavItemPage.php

```diff
@@ -446,19 +458,18 @@ public static function getPlaceholder($placeholderVar, $prevId, NavItemPage $nav
     }
 
     /**
-     * Get the arrayable values from a specific block id.
-     *
-     * @param integer $blockId
-     * @return array
+     * Get the arrayable values from a specific block.
+     * 
+     * @param array $envOptions Optional env options
      */
-    public static function getBlockItem(NavItemPageBlockItem $blockItem, NavItemPage $navItemPage)
+    public static function getBlockItem(NavItemPageBlockItem $blockItem, NavItemPage $navItemPage, $envOptions = null): array|false
     {
-        // if the block relation could be found, return false.
+        // if the block relation could not be found, return false.
         if (!$blockItem->block) {
             return false;
         }
 
-        $blockObject = $blockItem->block->getObject($blockItem->id, 'admin', $navItemPage);
+        $blockObject = $blockItem->block->getObject($blockItem->id, 'admin', $navItemPage->getNavItem());
```

Comment:
this won't pass the relation, this will pass an active query instance, not an active record. So if you like to pass the nav item object, then you have to use `navItemPage->navItem`. There is no unit tests for this, then this would fail here :-)

Reply:
I'm confused because of two reasons.
The guide https://luya.io/guide/cms/blocks.html#env-context-information says

> **pageObject**: Returns the `luya\cms\models\NavItem` object where you can run `luya\cms\models\NavItem->getNav()` to retrieve the `luya\cms\models\Nav` object.

And for the frontend context `getNavItem()` is already used in `NavItemPage::renderPlaceholderRecursive()`: https://github.com/luyadev/luya-module-cms/blob/a05aac2655ff4d1b62b875f97a00289aaad284d6/src/models/NavItemPage.php#L244

The pageObject should anyway be the same in frontend and admin context.